added extra ssl option for kibana

pescobar opened the pull request:
while using this role to deploy wazuh with ssl enabled we found error message `Kibana server is not ready yet` when trying to access the kibana webui in `https://kibana-host.com:5601/`

We enabled the kibana logs and found these error messages:

```
"type":"log","@timestamp":"2020-05-22T17:06:55Z","tags":["error","elasticsearch","data"],"pid":27536,"message":"Request error, retrying\nGET https://elastic-host.com:9200/_xpack => Hostname/IP does not match certificate's altnames: Host: elastic-host.com. is not in the cert's altnames: IP Address:192.168.1.200"}
```

After some googling we found [this post](https://github.com/elastic/kibana/issues/5305#issuecomment-451352008) and we fixed it by adding option `elasticsearch.ssl.verificationMode: certificate` to `/etc/kibana/kibana.yml` . This PR should add the fix to the default installation

jm404 commented:
Hi @pescobar ,

According to https://www.elastic.co/guide/en/kibana/current/settings.html, the default setting is `full` which checks the certificate and the hostname. Assuming you are using `elastic-host.com` it makes the certificate not to match the hostname which generates the error you see there.

So instead of forcing it to be `certificate`, `elasticsearch.ssl.verificationMode` should be customizable in order to allow users to choose between the different modes using a variable for example.

Best regards,

Jose

jm404 commented on roles/elastic-stack/ansible-kibana/templates/kibana.yml.j2:
```diff
@@ -110,6 +110,7 @@ elasticsearch.password: "{{ elasticsearch_xpack_security_password }}"
 server.ssl.enabled: true
 server.ssl.key: "{{node_certs_destination}}/{{ kibana_node_name }}.key"
 server.ssl.certificate: "{{node_certs_destination}}/{{ kibana_node_name }}.crt"
+elasticsearch.ssl.verificationMode: certificate
```
This should be customizable using a variable

pescobar commented:
thanks for your input and for taking into account my PR

I have updated the PR so this setting can be defined using a variable. I have also changed the default value to `full` instead of `certificate` so the previous default setting doesn't change. 

Please let me know if you want any other change.

p.s. I realized I was hitting this problem because I am not running kibana in the same host where elasticsearch is running and that's why hostnames in the certificate were not matching

jm404 commented:
Hi @pescobar,

Thanks for the quick turnaround!

We will run some tests on the change and then will ship it.

Best regards,

Jose
